### ISSUE TITLE
Heroes of the Storm Request

### ISSUE THREAD

**Issue by Jinyara** [antonpup/Aurora#105]
I saw what was done w/ the LoL profile and was wondering if it would be possible to get something similar for HotS?

**Comment by antonpup** [comment]
I briefly looked at HotS before, and it did not have any way of reading in-game information without either reading entire screen (witch is resource intensive) or memory reading (which will lead to a ban).

Sorry.

**Comment by ekyriakides** [comment]
@antonpup If you dont mind me asking (I couldn't find this information elsewhere) does Aurora hook into the game's SDK to retrieve information and also the Keyboard's SDK to update the colors thus making it safe to use (I assume that's what Razer Gaming Profiles do as well)?

**Comment by brainbug89** [comment]
No, Aurora don't hook into game's SDK except for Dota 2 and CS:GO (using GSI). The games itself sending RGB information to the corresponding devices of a manufacturer.

Aurora patches the libraries inside the game folders which sends the lighting information. This informations now get send to Aurora. Aurora then sends the received lighting information to the keyboards SDKs.

Rocket League is an exception. It don't support any RGB device, so Aurora is reading the memory to receive the needed information.

**Comment by ekyriakides** [comment]
So for example Aurora patches the Overwatch Razer library and sends it to Aurora than Razer? How dangerous is this?

**Comment by brainbug89** [comment]
> So for example Aurora patches the Overwatch Razer library and sends it to Aurora than Razer?
Yes.

> How dangerous is this?
See [FAQ](https://github.com/antonpup/Aurora#faq): "Support for all included games is tested to be sure not to trigger any anti-cheats or produce any suspicious behavior. With that said, the software is to be used at your own risk. We cannot prevent game developers from adding anti-cheats or blocking ways Aurora retrieves information from the game."

There is no known ban for using Aurora at any game yet. :wink:

**Comment by antonpup** [comment]
Not very (evidently), Aurora's wrappers don't load any additional dlls (thus most anti cheats don't detect any issues) but those anti cheats that check if file is signed (like h1z1) simply block the dll from loading.

I have yet to hear of anyone (myself included) getting banned in any game. I've been playing Overwatch, GTA 5, Battlefield 1, and Battlefield 3 all of which have anti cheats. None have triggered.

But with that said, I do not have any official statements regarding bannability of Aurora's wrappers. I did send Aurora's code to blizzard cheat team and such to make sure they look at it and know it exists. No reply from them.

**Comment by ekyriakides** [comment]
Thanks for the prompt replies guys. I've actually spoken to Blizzard CS who could not give me a clear answer, so I'll try and directly tweet maybe if they add the Corsair/Logitech SDK in their games (like in D3) it would be 100% safe to use.
